code/modules/mob/living/simple_mob/subtypes/animal/sif/grafadreka/grafadreka_interaction.dm

```diff
@@ -0,0 +1,117 @@
+/// Behaviors for being booped by drakes. If falsy, falls through to other behaviors.
+/atom/proc/interaction_grafadreka(mob/living/simple_mob/animal/sif/grafadreka/drake)

```

Comment:
Ultimately I chose this way because the interact_intent pattern puts the onus of knowledge on the actor rather than the actee; this might debase drake interaction back down to checking a_intent in handlers, but at least there's only one handler per thing and we can fall back to the existing interaction pattern if there's no special drake stuff to do.